### ISSUE TITLE
When restore raises an NU1301, build might fail with a `project.assets.json doesn't have a target for 'net6.0-windows10.0.19041.0` like error that's a red herring

### ISSUE THREAD

**Issue by nkolev92** [NuGet/Home#11862]
The real problem is that restore failed.

**Comment by nkolev92** [comment]
The improvement here is to create targets for the frameworks specified avoid this error. 
What might happen in that case is you get *a lot* of errors about missing types, more than you get now, but maybe that's a trade off we're willing to make?

**Comment by nkolev92** [comment]
If you build, then the restore error is gone and can't be brought back. 

Unsure what's going on here, this could be a thing that Nuget could improve, but it could be a bug with the project-system or SDK.

**Comment by michael-hawker** [comment]
@nkolev92 I hit this same cryptic message today as well.

For me, NuGet restore was failing as there was an invalid path/location/feed checked in my NuGet configuration options for Package Sources:

![image](https://user-images.githubusercontent.com/24302614/172508685-8fae1140-d957-46f1-b1bd-676ad5dc46f1.png)

With this checked, package restore was failing (as the directory didn't exist at the moment), but that wasn't output to the build log or even under the Package Manager output window. It was buried in the Error List along with a bunch of other compile errors and warnings, so I didn't find it for like 10 minutes to see the message that 'path could not be found' from nuget restore (forget the exact error code it gave there).

I've encountered this before when we changed our Azure DevOps feed name. Thought we filed an issue then, but can't find it now.

NuGet should either ignore invalid feeds (I mean if it can find the requested package elsewhere, why does this matter. And if it can't find a package then that would raise the issue to the developer who hopefully then sees something near there about the feed not being found), or it should better raise this error in configuration _before_ it gets to the build step as an explicit message output to the build output (and immediately fail the build).

Would save a lot of time and headaches.

**Comment by nkolev92** [comment]
> With this checked, package restore was failing (as the directory didn't exist at the moment), but that wasn't output to the build log or even under the Package Manager output window

That's what NU1301 does. The first restore puts the error in the error list, but for some reason the build is clearing these errors, unsure why that is as it's not NuGet that directly handles it, so I need to dig some more.

> It was buried in the Error List along with a bunch of other compile errors and warnings, so I didn't find it for like 10 minutes to see the message that 'path could not be found' from nuget restore (forget the exact error code it gave there).

I think that tends to be a function of running a build on top of a restore. Project-system has a tracking issue: https://github.com/dotnet/project-system/issues/3778, I'd appreciate if you can upvote that. 

> NuGet should either ignore invalid feeds (I mean if it can find the requested package elsewhere, why does this matter. And if it can't find a package then that would raise the issue to the developer who hopefully then sees something near there about the feed not being found), or it should better raise this error in configuration before it gets to the build step as an explicit message output to the build output (and immediately fail the build).

Starting 17.2/6.2/6.0.300 of VS, NuGet.exe and SDK, this gets raised by NuGet as you described. 

The thing that makes it problematic is that builds sometimes clear the restore errors, while they shouldn't. 😢 

On the topic of NuGet and source availablility, it is an intentional decision by NuGet to require sources to be available as it can affect resolution, https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#packagereference-and-sources.

**Comment by michael-hawker** [comment]
Thanks, upvoted other issue. I think the main problem I've encountered is that it's very unclear when a source isn't available that _that_ is the underlying issue beyond the message being buried (I don't think I saw anything about the path/source in my build output).

> On the topic of NuGet and source availablility, it is an intentional decision by NuGet to require sources to be available as it can affect resolution, https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#packagereference-and-sources.

That's good to know, and I understand that if I have a `nuget.config` dictating where the packages should be sourced from, then I would expect that if those sources aren't available, sure fail the build.

But if I have a global list of sources to look up to be able to pull in packages in VS and a matched package version is found on one source, and some other source isn't available. It's really confusing to have that fail my build. Like I'd much rather expect in the package explorer or as soon as I click build during the restore process to have an actual dialog in VS pop-up and say 'Hey, these source locations can't be contacted or no longer valid.' and give me a button to go to the options, the config file, and cancel.

As then it's an actionable thing I can understand and go fix vs. being confused about why my build is failing due to a transient or unknown issue with something I configured months ago in VS as a NuGet source.
